src/library/trackmodel.h

```diff
@@ -144,7 +146,20 @@ class TrackModel {
     virtual void select() {
     }
 
+    virtual int getPosition(){
+      return position;
+    }
+
+    virtual void setPosition(int newPosition){
+      if (newPosition <= 100 && newPosition >= 0 && abs(this->position-newPosition) < 10){
```

Comment:
this-> is redundant. Please add a m_ prefix to position to indicate it as member variable.